src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -104,25 +119,34 @@ public function reverseTransform($value)
         return $collection;
     }
 
-    public function transform($entityOrCollection)
+    /**
+     * @param object|object[]|null $value
+     *
+     * @return array<string|int, string>
+     *
+     * @phpstan-param T|T[]|null $value
+     *
+     * @phpstan-return array{_labels?: mixed}
+     */
+    public function transform($value)
     {
         $result = [];
 
-        if (!$entityOrCollection) {
+        if (!$value) {
             return $result;
         }
 
         if ($this->multiple) {
-            $isArray = \is_array($entityOrCollection);
-            if (!$isArray && substr(\get_class($entityOrCollection), -1 * \strlen($this->className)) === $this->className) {
+            $isArray = \is_array($value);
+            if (!$isArray && substr(\get_class($value), -1 * \strlen($this->className)) === $this->className) {
                 throw new \InvalidArgumentException(
                     'A multiple selection must be passed a collection not a single value.'
                     .' Make sure that form option "multiple=false" is set for many-to-one relation and "multiple=true"'
                     .' is set for many-to-many or one-to-many relations.'
                 );
             }
-            if ($isArray || ($entityOrCollection instanceof \ArrayAccess)) {
-                $collection = $entityOrCollection;
+            if ($isArray || ($value instanceof \ArrayAccess)) {
+                $collection = $value;
             } else {
                 throw new \InvalidArgumentException(
```

Comment:
There could be some improvements in the check to avoid throwing the same exception here and in the previous `if`.

Reply:
This method is also very long and should be refactored. This should be done in another PR.

Reply:
Yes, I didn't considered this as a blocking point for the merge.